examples/source_examples/newsapi/connector.py

```diff
@@ -0,0 +1,197 @@
+
+# See the Technical Reference documentation (https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update)
+# and the Best Practices documentation (https://fivetran.com/docs/connectors/connector-sdk/best-practices) for details
+
+# Import requests to make HTTP calls to API
+import requests as rq
+import traceback
+import datetime
+import json
+
+# Import required classes from fivetran_connector_sdk
+from fivetran_connector_sdk import Connector
+from fivetran_connector_sdk import Logging as log
+from fivetran_connector_sdk import Operations as op
+
+# Define the schema function which lets you configure the schema your connector delivers.
+# See the technical reference documentation for more details on the schema function:
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#schema
+# The schema function takes one parameter:
+# - configuration: a dictionary that holds the configuration settings for the connector.
+def schema(configuration: dict):
+
+    if not ('topic' in configuration):
+        raise ValueError("Could not find 'topic' in configuration.json")
+    
+    return [
+        {
+            "table": "article",
+            "primary_key": ["source", "published_at"],
+            "columns": {
+                "source": "STRING",
+                "published_at": "UTC_DATETIME",
+                "author": "STRING",
+                "title": "STRING"
+            }
+        }
+    ]
+
+# Define the update function, which is a required function, and is called by Fivetran during each sync.
+# See the technical reference documentation for more details on the update function
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
+# The state dictionary is empty for the first sync or for any full re-sync
+def update(configuration: dict, state: dict):
+
+    try:
+        conf = configuration
+        base_url = "https://newsapi.org/v2/everything"
+        from_ts = state['to_ts'] if 'to_ts' in state else '2024-11-10T00:00:00'
```

Comment:
Default value can be set 1 day ago

from_ts = state['to_ts'] if 'to_ts' in state else datetime.datetime.now() - datetime.timedelta(days=1)

Reply:
this is because default API key only allows 1 day of data